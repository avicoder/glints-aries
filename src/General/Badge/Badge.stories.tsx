import React from 'react';
import { Story, Meta } from '@storybook/react';

import { Badge, Props } from './Badge';
import { BaseContainer } from '../../Layout/GlintsContainer/GlintsContainer';
import styled from 'styled-components';

const BadgeStack = styled('div')`
  display: flex;
  align-items: flex-start;
`;

export default {
  title: 'General/Badge',
  component: Badge,
  decorators: [Story => <BaseContainer>{Story()}</BaseContainer>],
} as Meta;

const Template: Story<Props> = args => <Badge {...args}></Badge>;

export const Interactive = Template.bind({});

Interactive.args = {
  label: 'hello',
};

export const Variants: Story<Props> = () => (
  <BadgeStack>
    <Badge label="4" variant="default" />
    <Badge label="4" variant="dimmed" />
    <Badge label="4" variant="white" />
    <Badge label="4" variant="blue" />
  </BadgeStack>
);
