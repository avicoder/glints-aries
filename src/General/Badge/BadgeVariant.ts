export const BadgeVariant = {
  DEFAULT: 'default',
  DIMMED: 'dimmed',
  WHITE: 'white',
  BLUE: 'blue',
};
