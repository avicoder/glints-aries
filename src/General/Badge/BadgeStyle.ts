import styled from 'styled-components';
import { PrimaryColor, Greyscale } from '../../Utils/Colors';
import { BadgeVariant } from './BadgeVariant';

export const BadgeContainer = styled.div<BadgeContainerProps>`
  position: relative;
  display: inline-flex;
  background: ${({ variant }) =>
    variant === BadgeVariant.DEFAULT
      ? `${PrimaryColor.glintsred}`
      : `${Greyscale.grey}`};
  color: ${Greyscale.white};
  border-radius: 3px;
  margin: 0 4px;

  span {
    height: 17px;
    min-width: 19px;
    font-size: 12px;
    font-weight: 700;
    padding: 0 5px;
    text-align: center;
  }
`;

interface BadgeContainerProps {
  variant?: string;
}
