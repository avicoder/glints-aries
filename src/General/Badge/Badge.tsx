import * as React from 'react';
import classNames from 'classnames';
import isNil from 'lodash/isNil';
import { BadgeContainer } from './BadgeStyle';
import { BadgeVariant } from './BadgeVariant';

export const Badge: React.FunctionComponent<Props> = ({
  label,
  variant = BadgeVariant.DEFAULT,
  className,
  ...defaultProps
}) => (
  <React.Fragment>
    {!isNil(label) && (
      <BadgeContainer
        className={classNames('aries-badge', className)}
        role="presentation"
        variant={variant}
        {...defaultProps}
      >
        <span className="badge-content">{label}</span>
      </BadgeContainer>
    )}
  </React.Fragment>
);

export type BadgeType = 'dimmed' | 'default' | 'white' | 'blue';

export interface Props
  extends React.ComponentPropsWithoutRef<typeof BadgeContainer> {
  /** Sets the label of Badge. */
  label: string | number;
  /** Sets the variant of the Badge. */
  variant?: BadgeType;
}

export default Badge;
