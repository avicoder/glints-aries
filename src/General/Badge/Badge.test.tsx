import * as React from 'react';

import 'jest-styled-components';
import * as renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

import Badge, { BadgeType } from './Badge';

const label = '6';

it(`<Badge> should render a label with the text "6"`, () => {
  const BadgeSnapshot = renderer.create(<Badge label={label} />).toJSON();
  expect(BadgeSnapshot).toMatchSnapshot();
});

it('should display the correct text when rendered', () => {
  const { queryByText } = render(<Badge label={label} />);
  const BadgeContent = queryByText(label);
  expect(BadgeContent).toBeVisible();
});

describe('when sup is:', () => {
  it('true, should have a top property', () => {
    const { getByRole } = render(<Badge label={label} sup />);
    const badgeContainer = getByRole('presentation');
    expect(badgeContainer).toHaveStyle('top: -.6em');
  });

  it('false, should not have a top property', () => {
    const { getByRole } = render(<Badge label={label} sup={false} />);
    const badgeContainer = getByRole('presentation');
    const { top } = getComputedStyle(badgeContainer);
    expect(top).toBeFalsy();
  });
});

describe('<Badge/> snapshots with variant prop', () => {
  const matchSnapshotWithVariant = (variant: BadgeType) => {
    test(`variant ${variant}`, () => {
      const { asFragment } = render(<Badge variant={variant} label={label} />);
      expect(asFragment()).toMatchSnapshot();
    });
  };

  ['default', 'dimmed'].forEach((variant: BadgeType) =>
    matchSnapshotWithVariant(variant)
  );
});
