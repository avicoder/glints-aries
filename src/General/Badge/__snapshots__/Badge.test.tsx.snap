// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge/> snapshots with variant prop variant blue 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  background: #017EB7;
  color: #FFFFFF;
  border-radius: 3px;
  margin: 0 4px;
  padding: 0 4px;
  height: 17px;
  min-width: 18px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 span {
  font-size: 12px;
  font-weight: 700;
  text-align: center;
}

<div
    class="c0 aries-badge"
    role="presentation"
  >
    <span
      class="badge-content"
    >
      6
    </span>
  </div>
</DocumentFragment>
`;

exports[`<Badge/> snapshots with variant prop variant default 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  background: #ed9300;
  color: #FFFFFF;
  border-radius: 3px;
  margin: 0 4px;
  padding: 0 4px;
  height: 17px;
  min-width: 18px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 span {
  font-size: 12px;
  font-weight: 700;
  text-align: center;
}

<div
    class="c0 aries-badge"
    role="presentation"
  >
    <span
      class="badge-content"
    >
      6
    </span>
  </div>
</DocumentFragment>
`;

exports[`<Badge/> snapshots with variant prop variant dimmed 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  background: #666666;
  color: #FFFFFF;
  border-radius: 3px;
  margin: 0 4px;
  padding: 0 4px;
  height: 17px;
  min-width: 18px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 span {
  font-size: 12px;
  font-weight: 700;
  text-align: center;
}

<div
    class="c0 aries-badge"
    role="presentation"
  >
    <span
      class="badge-content"
    >
      6
    </span>
  </div>
</DocumentFragment>
`;

exports[`<Badge/> snapshots with variant prop variant white 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  background: #FFFFFF;
  color: #017EB7;
  border-radius: 3px;
  margin: 0 4px;
  padding: 0 4px;
  height: 17px;
  min-width: 18px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 span {
  font-size: 12px;
  font-weight: 700;
  text-align: center;
}

<div
    class="c0 aries-badge"
    role="presentation"
  >
    <span
      class="badge-content"
    >
      6
    </span>
  </div>
</DocumentFragment>
`;

exports[`<Badge> should render a label with the text "6" 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  background: #ed9300;
  color: #FFFFFF;
  border-radius: 3px;
  margin: 0 4px;
  padding: 0 4px;
  height: 17px;
  min-width: 18px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 span {
  font-size: 12px;
  font-weight: 700;
  text-align: center;
}

<div
  className="c0 aries-badge"
  role="presentation"
>
  <span
    className="badge-content"
  >
    6
  </span>
</div>
`;
